obj/generators.h

```diff
@@ -46,8 +46,8 @@ namespace generators
     public:
         PointSource() = delete;
         explicit PointSource::PointSource(const double &freq, const size_t &numOfRaysPerRow, \
-                                        const double &diffusorSize, RandomGen* randomGen) : _frequency(freq),\
-                                        _numOfRaysPerRow(numOfRaysPerRow), _sampleSize(diffusorSize), \
+                                        const double &SampleSize, RandomGen* randomGen) : _frequency(freq),\
```

Comment:
Do not use const references to primitive types. There is literally no benefit to it, since the amount of time it takes to copy a float is the same as to copy a reference pointer.

Reply:
I need to fix entire project for that ;P

Reply:
One file at a time.

Reply:
More info why doing this is a bad idea: https://stackoverflow.com/questions/14013139/is-it-counter-productive-to-pass-primitive-types-by-reference

This can even incur some performance cost.

FME: have you read somewhere to use const ref for primitive types? Is this in the book you read?

Reply:
addded to my TODO's... I dint read about it in the book. Maybe I missed something, but for now I will be aware of it ;)